fix two cases where site_url() was used instead of home_url()

Levdbas opened the pull request:
**Ticket**: #2356 

## Issue
site_url() returns a wrong url on bedrock based installations, since site_url() returns https://yourdomain.com/wp since the WP installation lives in this directory.

Relevant file structure of a Bedrock based project:

├── Other stuff, no relevant for this PR.
└── web                        → Web root (document root on your webserver)
    ├── app                    → **wp-content equivalent**
    │   ├── mu-plugins   → Must use plugins
    │   ├── plugins         → Plugins
    │   ├── themes         → Themes
    │   └── uploads         → Uploads
    ├── wp-config.php    → Required by WP (never edit)
    ├── index.php           → WordPress view bootstrapper
    └── wp                     → **WordPress core**

## Solution
When using home_url() this issue is resolved.

## Impact
In my initial issue I found out that changing `site_url()` to `home_url()` in  `Timber\URLHelper::file_system_to_url()` fixed a sideloading img issue. This change is inside PR. While on it I found a second instance of `site_url` in `Timber\ImageHelper::_get_file_url`. to get a relative URL we need to remove `home_url()` from the $url instead of `site_url()` since uploads in a Bedrock based installation live in `https://yourdomain.com/app/uploads/` and not in `https://yourdomain.com/wp/app/uploads/`

## Considerations
It might be a good idea to write an extra set of tests based off a Bedrock based installation because I see the combination of Bedrock + Timber a lot in my freelance work.

jarednova commented:
Thanks @Levdbas ! looks like there's just one failing test we need to track down ...

```php
1) TestTimberURLHelper::testFileSystemToURLWithWPML
Failed asserting that 'http://example.org/wp-content/uploads/2020/11/arch.jpg' ends with "://example2.org/wp-content/uploads/2020/11/arch.jpg".
/home/travis/build/timber/timber/tests/test-timber-url-helper.php:62
```

Levdbas commented:
Hi @jarednova , I am kinda at a stale here... Tried for many hours but can't find a proper solution. 

Problem is as follows:

home_url() works in default and bedrock setups **without** WPML. 
site_url() works in default WP **with** WPML but not in Bedrock, because of the issues i started this issue anyway.
home_url() **does not** work **with** WPML, regardless of the setup.

WPML seem to hijack the home_url() function and add the home url for the current language.

So using home_url() is not an option because of WPML, and site_url() is not an option because of Bedrock. I am looking for a way to just get the base home url without the language and without any subdir where WP is living. And I cant find one. :(

Any ideas how we can work around this issue?

jarednova commented:
Oy. Always something, huh @Levdbas?

My thought is two steps ...

1. Because using `home_url()` seems like the correct "base" behavior the Timber code should be updated to begin there.
2. Okay, so now what about WPML? Like you said, it's doing funny stuff so let's isolate that into its own integrations file (like we have for ACF and a few others) that can hook into `ImageHelper` and `URLHelper` where needed. No doubt it might involve some URL acrobatics, but at least it will only be in play for people who are using a combo of those two plug-ins in tandem (not nobody, but certainly a vast minority of the user base)
3. If we get stuck, we've received help in the past from the WPML team to figure out what special sauce is needed to play nice (also seems like [an issue others have encountered/resolved](https://github.com/roots/bedrock/issues/30))

Levdbas commented:
Hi @jarednova , it was the test after all 🥇 . Updated the test case in the first commit.
in f38aa2f I've also added the specific WPML filter so ImageHelper keeps on working with WPML. Could you take a look if i applied the filters corretly this way with the right naming scheme?

jarednova commented:
Holy moly! Great stuff @Levdbas. I'll take a closer look this week to make sure the filter application is correct and keeping with naming, etc.

jarednova reviewed (approved):
The only thing worth considering is whether the preceding underscore in the filter name is weird. It is. But it matches with the format we use elsewhere, and not worth creating a one-off "rule" that omits it here.

Maybe in the future we'll clean up the function names in `ImageHelper` and a few other locations to remove those underscored functions. But let's not distract us from making this fix today.

gchtr commented:
> Maybe in the future we'll clean up the function names in `ImageHelper` and a few other locations to remove those underscored functions. But let's not distract us from making this fix today.

@jarednova As soon as we tackle coding standards, the underscores prefixes will go away.

jarednova commented:
👍  sounds good! much better to take those things on as a whole vs. ad-hoc bits here-and-there
